packages/backend/src/Controllers/Election/getElectionResultsController.ts

```diff
@@ -31,17 +31,37 @@ const getElectionResults = async (req: IElectionRequest, res: Response, next: Ne
     const election = req.election
     let results: ElectionResults[] = []
     for (let race_index = 0; race_index < election.races.length; race_index++) {
-        const candidateNames = election.races[race_index].candidates.map((Candidate: any) => (Candidate.candidate_name))
-        const race_id = election.races[race_index].race_id
+        const race = election.races[race_index]
+        const candidateNames = race.candidates.map((Candidate: any) => (Candidate.candidate_name))
+        const candidateIDs = race.candidates.map((Candidate: any) => (Candidate.candidate_id))
+        const useWriteIns = race.enable_write_in && race.write_in_candidates && race.write_in_candidates.length>0
+        const writeInCandidates = useWriteIns && race.write_in_candidates ? race.write_in_candidates : [] 
+        const writeInCandidateNames = useWriteIns && race.write_in_candidates ? race.write_in_candidates.map(candidate => candidate.candidate_name) : []
+        const number_of_candidates = candidateNames.length + writeInCandidateNames.length
+        const race_id = race.race_id
         const cvr: ballot[] = []
-        const num_winners = election.races[race_index].num_winners
-        const voting_method = election.races[race_index].voting_method
+        const num_winners = race.num_winners
+        const voting_method = race.voting_method
+        const fullCandidateNames = [...candidateNames,...writeInCandidateNames]
+
         ballots.forEach((ballot: Ballot) => {
             const vote = ballot.votes.find((vote) => vote.race_id === race_id)
             if (vote) {
-                let row: ballot = vote.scores.map((score: Score) => (
-                    score.score
-                ))
+                let row: ballot = new Array(number_of_candidates).fill(null)
+                vote.scores.forEach(score => {
+                    const candidateIndex = candidateIDs.indexOf(score.candidate_id)
+                    if (candidateIndex >= 0) {
+                        row[candidateIndex] = score.score
+                    }
+                    else if (useWriteIns && score.write_in_name) {
+                        const write_in_name = score.write_in_name // typescript sees score.write_in_name as possibly undefined unless I extract it to another variable for some reason
+                        const writeInCandidateIndex = writeInCandidates.findIndex(candidate => candidate.aliases.includes(write_in_name) && candidate.approved)
+                        if (writeInCandidateIndex >= 0) {
+                            row[writeInCandidateIndex + candidateNames.length] = score.score
+                        }
```

Comment:
Should we add an else clause to throw (or log) a warning?

Feels like the writeInCandidateIndex is expected to always resolve

Reply:
It wouldn't in the case that the write in candidate wasn't approved by the admin. But maybe that should be a separate check? First check if they are an alias to a candidate, if not throw error and admin can revisit write in section. Then check if approved, then add score.

Reply:
Actually, maybe it would be better to have a counter for unprocessed write ins. That way the results can still be retrieved live and display with the results that X write in names are still unprocessed.

Reply:
Updated results to have an unprocessed write in candidate count

Reply:
Nice! Yeah I was just thinking that we should have some default write-in behavior in addition to allowing the admin to configure the aliases

Counting each new variation as it's own candidate is a good starting point, but I could also see us performing some minimal typo checks by default.